test/testlib/nuodb_database_utilities.go

```diff
@@ -275,7 +275,13 @@ func UpgradeDatabase(t *testing.T, namespaceName string, helmChartReleaseName st
 	if upgradeOptions.TePodShouldGetRecreated {
 		AwaitPodObjectRecreated(t, namespaceName, tePod, 30*time.Second)
 	}
-	AwaitPodUp(t, namespaceName, tePodName, 180*time.Second)
+	// the TE ReplicaSet can be recreated with a different name so fetch the pod
+	// using a template continuously
+	Await(t, func() bool {
+		pod, err := findPod(t, namespaceName, tePodNameTemplate)
+		require.NoError(t, err, tePodNameTemplate)
```

Comment:
Is it correct to do an assertion in this function that is returning a bool indicating if the check was successful?

Reply:
Good point! We don't need to fail fast in this case.